dvc/tree/base.py

```diff
@@ -330,7 +330,14 @@ def get_dir_hash(self, path_info, **kwargs):
         hash_info.size = dir_info.size
         return hash_info
 
-    def upload(self, from_info, to_info, name=None, no_progress_bar=False):
+    def upload(
+        self,
+        from_info,
+        to_info,
+        name=None,
+        no_progress_bar=False,
+        file_mode=None,
```

Comment:
`file_mode` is already supported in `BaseTree.download()`, so I think specifying mode here fits into the existing tree API